orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/appengine/AppEngineServerGroupCreator.groovy

```diff
@@ -47,4 +53,43 @@ class AppEngineServerGroupCreator implements ServerGroupCreator {
   Optional<String> getHealthProviderName() {
     return Optional.empty()
   }
+
+  static private void resolveArtifacts(Stage stage, Map operation) {
+    if (operation.repositoryUrl) {
+      return
+    }
+
+    Map expectedArtifact = operation.expectedArtifact
+    if (operation.fromArtifact && expectedArtifact && expectedArtifact.fields) {
+      Execution execution = stage.getExecution()
+      Map<String, Object> trigger = [:]
+      if (execution instanceof Pipeline) {
+        trigger = ((Pipeline) execution).getTrigger()
+      }
+
+      List<Map> artifacts = (List<Map>) trigger.artifacts
+      Artifact artifact = (Artifact) artifacts.find { a -> ArtifactResolver.expectedMatch((ExpectedArtifact) expectedArtifact, (Artifact) a) }
+      if (artifact?.reference) {
+        String repositoryUrl = ''
+        switch (artifact.type) {
+          // TODO(jacobkiefer): These object types are pretty fragile, we need to harden this somehow.
```

Comment:
I'm guessing this will all be moved to clouddriver? e.g. this this passes the artifact along w/o much question

Reply:
Yes, we can do that refactor in the future.

Reply:
Sounds good